_posts/2023-03-21-integrating-foreman-with-event-driven-ansible.md

```diff
@@ -0,0 +1,229 @@
+---
+layout: post
+title: Integrating Foreman with Event-Driven Ansible
+date: 2023-03-21  8:19:32
+author: Evgeni Golov
+tags:
+- foreman
+- ansible
+---
+
+[Event-Driven Ansible](https://www.ansible.com/use-cases/event-driven-automation) is a new way of automation where your automation can dynamically react to events that happen in your environment.
+Today, we want to show you how you can integrate that with Foreman using the [Foreman Webhooks plugin](https://github.com/theforeman/foreman_webhooks).
+
+<!--more-->
+
+As workflows and events one might want to react on will vary across different setups, we'll show one example integration and let everyone adopt it to their needs.
+The workflow in this example is to have machines in a group update once their content has been updated in Katello.
+
+First of all, we need a Katello installation with the Foreman Webhooks plugin enabled.
+This can be achieved by calling `foreman-installer --enable-foreman-plugin-webhooks` on an existing setup or passing `--enable-foreman-plugin-webhooks` during the initial installation.
+
+Second, we need a place to run Ansible and Ansible-Rulebook.
+An empty CentOS Stream 8 machine where we follow the [ansible-rulebook installation](https://ansible-rulebook.readthedocs.io/en/stable/installation.html) and the [ansible.eda collection installation](https://github.com/ansible/event-driven-ansible) guides will do:
+
+```console
+[root@eda] # dnf install java-17-openjdk python39-pip
+[user@eda ~]$ export JAVA_HOME=/usr/lib/jvm/jre-17-openjdk
+[user@eda ~]$ python3.9 -m venv ./eda-venv
+[user@eda ~]$ . ./eda-venv/bin/activate
+(eda-venv) [user@eda ~]$ pip install ansible-rulebook ansible ansible-runner
+(eda-venv) [user@eda ~]$ ansible-galaxy collection install ansible.eda
+(eda-venv) [user@eda ~]$ pip install -r /home/user/.ansible/collections/ansible_collections/ansible/eda/requirements.txt
+```
+The only difference to the official guides is that we explicitly used Python 3.9, as the default Python in CentOS Stream 8 is Python 3.6 and that is too old for `ansible-rulebook` and `ansible-runner`.
+
+When we now try to run the `ansible-rulebook` CLI, it will fail due to a [bug in the 0.11.0 release](https://github.com/ansible/ansible-rulebook/issues/455):
```

Comment:
I've been told there will be a release soon, so if we publish the post *after* the release, we can avoid the odd hacks in here.